fladle-plugin/src/main/java/com/osacky/flank/gradle/FladlePluginDelegate.kt

```diff
@@ -106,6 +107,10 @@ class FladlePluginDelegate {
         environment(mapOf("GOOGLE_APPLICATION_CREDENTIALS" to config.serviceAccountCredentials.get()))
       }
       dependsOn(writeConfigProps)
+      if (config.dependOnAssemble.isPresent && config.dependOnAssemble.get()) {
+        dependsOn(ASSEMBLE_ANDROID_TEST)
+        dependsOn("assemble")
```

Comment:
I don't think this should depend on assemble. That would build both debug and release variants. In this case we likely only want the debug variant.

We also want to make sure that this works for all variants. Can you please add a test which tests that the proper task dependencies are executed?

Reply:
Using `assembleProvider` to get appropriate assemble task now. Will add test!